site/docs/v1/tech/connectors/index.adoc

```diff
@@ -35,12 +35,34 @@ In contrast to using an OIDC Identity Provider, with a Connector the user doesn'
 
 Using a Connector for authentication allows you to use an external user management system as another data source for FusionAuth. If the data source can speak LDAP or HTTP, you can authenticate users against it.
 
+==== Required fields
+
+If you are authenticating a user, you must provide the following fields in the user object you return.
+
+* `user.username` or `user.email`
+* `user.id`: a FusionAuth compatible GUID
+
+If you are using the generic connector, you are building the `user` object, so ensure these fields are in the JSON. If you are using the LDAP connector, make sure you set these fields in the lambda.
+
+In addition, ensure that the same `user.id` is returned each time a user is retrieved. If you have to generate the GUID, return the same one each time the user authenticates. This ensures that FusionAuth knows which user is being referenced, even if the username or email address changes over time.
+
 === Migration
 
 In this scenario, the user data is migrated from the external data source to FusionAuth. Subsequent logins will authenticate against FusionAuth, not the external data store. In addition, any changes in the external data store will not be propagated to FusionAuth. 
 
 Using a Connector in this way allows for a phased migration. Let your users sign in and migrate their data as they do so. All changes to user data like group membership should then be made in FusionAuth. You can run the old system for a time, then shut it off and remove its configuration, then relying on FusionAuth for all user authentication.
 
+==== Required fields
+
+If you are migrating a user, you must provide the following fields in the user object you return.
+
+* `user.username` or `user.email`
+* `user.id`: a FusionAuth compatible GUID
+
+If you are using the generic connector, you are building the `user` object, so ensure these fields are in the JSON. If you are using the LDAP connector, make sure you set these fields in the lambda.
+
+If you don't have a GUID to associate with this user, you may create a random one, as the source datastore won't be consulted after the user is initially migrated. 
```

Comment:
I would say we should always use the term UUID unless we are referencing a specific data type in a third party system such as AD.